curve_apps/edge_detection/params.py

```diff
@@ -72,11 +83,19 @@ class Parameters(BaseData):
 
     _name: str = NAME
 
-    input_file: InputFile | None = InputFile.read_ui_json(
-        DEFAULT_UI_JSON, validate=False
-    )
+    input_file: InputFile = InputFile.read_ui_json(DEFAULT_UI_JSON, validate=False)
     detection: DetectionParameters
     output: OutputParameters
     run_command: str = f"curve_apps.{NAME}.driver"
     source: SourceParameters
     title: str = NAME.capitalize().replace("_", " ")
+
+    @model_validator(mode="after")
+    def update_input_file(self):
+        if self.input_file is None or not self.input_file.validate:
+            params_data = self.flatten()
+            if self.input_file.ui_json is None:
```

Comment:
Also don't think that's possible here since we are loading it with defaults. Better would be to validate on "before"

Reply:
it's just there for mypy